src/print.jl

```diff
@@ -324,6 +326,175 @@ function objective_function_string(print_mode, model::Model)
    end
 end
 
+struct _SolutionSummary
+    verbose::Bool
+    solver::String
+    # Status
+    termination_status::MOI.TerminationStatusCode
+    primal_status::MOI.ResultStatusCode
+    dual_status::MOI.ResultStatusCode
+    raw_status::String
+    result_count::Int
+    has_values::Bool
+    has_duals::Bool
+    # Candidate solution
+    objective_value::Float64
+    objective_bound::Union{Missing,Float64}
+    dual_objective_value::Union{Missing,Float64}
+    primal_solution::Dict{String,Float64}
+    dual_solution::Dict{String,Float64}
+    # Work counters
+    solve_time::Float64
+    barrier_iterations::Union{Missing,Int}
+    simplex_iterations::Union{Missing,Int}
+    node_count::Union{Missing,Int}
+end
+
+"""
+    solution_summary(model::Model; verbose::Bool = false)
+
+Return a struct that can be used print a summary of the solution.
+
+If `verbose=true`, write out the primal solution for every variable and the
+dual solution for every constraint, excluding those with empty names.
+
+## Examples
+
+When called at the REPL, the summary is automatically printed:
+```julia
+julia> solution_summary(model)
+[...]
+```
+
+Use `print` to force the printing of the summary from inside a function:
+```julia
+function foo(model)
+    print(solution_summary(model))
+    return
+end
+```
+"""
+solution_summary(model::Model; verbose::Bool = false) = _SolutionSummary(
+    verbose,
+    solver_name(model),
+    termination_status(model),
+    primal_status(model),
+    dual_status(model),
+    raw_status(model),
+    result_count(model),
+    has_values(model),
+    has_duals(model),
+    objective_value(model),
+    _try_get(objective_bound, model),
+    _try_get(dual_objective_value, model),
+    _get_solution_dict(model),
```

Comment:
Maybe only do this in case `verbose` is `true` ?

Reply:
Done